src/tools/cephfs/cephfs-shell

```diff
@@ -466,7 +466,10 @@ class CephFSShell(Cmd):
         except (libcephfs.Error, Exception) as e:
             if shell.debug:
                 traceback.print_exc(file=sys.stdout)
-            set_exit_code_msg(msg=e)
+            if type(e).__name__ == 'Cmd2ArgparseError':
```

Comment:
Why is this exception ignored ?

Reply:
The existing code `set_exit_code_msg(msg=e)` in this block of code prints just the exception message without the type of exception that occurred. When any cephfs-shell command is run with an unexpected number of args, for e.g. `get a b c`, it will show  `get: error: unrecognized arguments: c` followed by a blank line (this actually is `set_exit_code_msg(msg=e)` but as `e` is empty in the case of `Cmd2ArgparseError` that blank line exists) but now as the type of exception is also going to be printed with the exception message using `set_exit_code_msg(msg=f'{type(e).__name__}: {e}')`, that blank line will be replaced by `Cmd2ArgparseError:` which is not required as it already will show (in our example) => `get: error: unrecognized arguments: c`  and therefore this exception is inessential and thus safe to ignore . Not only it's not required but also getting `Cmd2ArgparseError:` everytime it encounters wrong number of args looks very dirty.

Reply:
Ack

Reply:
Better to add a note for this here (just before the `pass`) so that the reasoning is recorded for future since it's not immediately apparent?

```
# NOTE: In case of Cmd2ArgparseError the error message is already printed beforehand (plus Cmd2ArgparseError instances have empty message)
```